lib/auth/auth.go

```diff
@@ -4582,8 +4615,22 @@ func (a *Server) GetTokens(ctx context.Context, opts ...services.MarshalOption)
 	return tokens, nil
 }
 
-// NewWebSession creates and returns a new web session for the specified request
-func (a *Server) NewWebSession(ctx context.Context, req types.NewWebSessionRequest) (types.WebSession, error) {
+// newWebSessionOpts are WebSession creation options exclusive to Auth.
+// These options complement [types.NewWebSessionRequest].
+// See [Server.newWebSession].
+type newWebSessionOpts struct {
+	// deviceExtensions are the device extensions to apply to the session.
+	// Only present on renewals, the original extensions are applied by
+	// [Server.AugmentWebSessionCertificates].
```

Comment:
Any sensible way to consolidate this? As in, have `AugmentWebSessionCertificates` and `newWebSession` call the same thing? I don't understand this section of the code well, just a "spidey sense" that maybe you could simplify something here, could be totally wrong though.

Reply:
This is used as the input for `a.generateUserCert`, so there isn't much more to simplify. AugmentWebSessionCertificates does a whole lot of verifications before letting the extensions end up in the certificate, but during renewal we just pass them along. Other kinds of device trust renewals (think MFA certificates) behave similarly.

I hope this clarifies a bit.